crates/consensus-any/src/transaction/unknown/typed.rs

```diff
@@ -0,0 +1,198 @@
+use std::sync::OnceLock;
+
+use crate::AnyTxType;
+use alloy_eips::{eip2930::AccessList, eip7702::SignedAuthorization};
+use alloy_primitives::{Address, Bytes, ChainId, TxKind, B256, U128, U256, U64};
+use alloy_serde::OtherFields;
+
+/// Memoization for deserialization of [`UnknownTxEnvelope`],
+/// [`UnknownTypedTransaction`] [`AnyTxEnvelope`], [`AnyTypedTransaction`].
+/// Setting these manually is discouraged, however the fields are left public
+/// for power users :)
+///
+/// [`AnyTxEnvelope`]: crate::AnyTxEnvelope
+/// [`AnyTypedTransaction`]: crate::AnyTypedTransaction
+/// [`UnknownTxEnvelope`]: crate::UnknownTxEnvelope
+#[derive(Debug, Clone, PartialEq, Eq, Default)]
+#[allow(unnameable_types)]
+pub struct DeserMemo {
+    pub input: OnceLock<Bytes>,
+    pub access_list: OnceLock<AccessList>,
+    pub blob_versioned_hashes: OnceLock<Vec<B256>>,
+    pub authorization_list: OnceLock<Vec<SignedAuthorization>>,
+}
+
+/// A typed transaction of an unknown Network
+#[derive(Debug, Clone, PartialEq, Eq)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+#[doc(alias = "UnknownTypedTx")]
+pub struct UnknownTypedTransaction {
+    #[cfg_attr(feature = "serde", serde(rename = "type"))]
+    /// Transaction type.
+    pub ty: AnyTxType,
+
+    /// Additional fields.
+    #[cfg_attr(feature = "serde", serde(flatten))]
+    pub fields: OtherFields,
+
+    /// Memoization for deserialization.
+    #[cfg_attr(feature = "serde", serde(skip, default))]
+    pub memo: DeserMemo,
```

Comment:
Due to the migration of `UnknownTypedTransaction` to `consensus-any`, a std-based implementation is now required in `consensus-any`, but should we change OnceCell to a no_std implementation

Reply:
Changed to export this only if `std` feature is enabled: https://github.com/alloy-rs/alloy/pull/1715/commits/fd43c12537adceaf5c606dcb4631a42f6c0dfc8b